### ISSUE TITLE
Every new release breaks deployed apps

### ISSUE THREAD

**Issue by gyzerok** [thorvg/thorvg.web#82]
Hello there!

Every new release you are making is causing us breakage before we update to it. This is because you are pointing to the latest version in your code https://github.com/thorvg/thorvg.web/blob/e189798bfd04d5a8e98a49a7c33fd5973a15bc3b/src/lottie-player.ts#L34

The best solution would be of course to use local copy of the wasm file from `node_modules`, so projects do not depend on 3rd-party cdn and have reproducible builds. But even using appropriate version from `unpkg` would already go a long way.

Thank you!

**Comment by tinyjin** [comment]
@gyzerok Yes, we were aware of this point. That's why we introduce wasmUrl property #80. When using CDN, it'd be hard to ensure consistent loading time and reproducible builds as you mentioned.

The `_wasmUrl` should be fallback link, and users should be able to handle origin of WASM binary.
The update enabling custom WasmURL will soon be shipped. Would it be answer for you?

**Comment by gyzerok** [comment]
@tinyjin yeah, thank you! I'll wait for the next version, check that everything works with and will close the issue if so.

**Comment by tinyjin** [comment]
@gyzerok ThankPlz check this ;)

Custom WASM URL has just shipped to `v1.0.0-pre.9`

### Web Component

```html
<lottie-player
      autoPlay
      loop
      src="https://lottie.host/6d7dd6e2-ab92-4e98-826a-2f8430768886/NGnHQ6brWA.json"
      wasmUrl="/dist/thorvg-wasm.wasm"
      style="width: 300px; height: 300px"
></lottie-player>
```

### JS-creation

```js
player = document.createElement('lottie-player');
player.autoPlay = true;
player.loop = true;
player.wasmUrl = '/thorvg-wasm.wasm'; // Set Custom WASM URL
player.renderConfig = { renderer };
```

### React/NextJS

https://github.com/thorvg/thorvg.web/commit/bedd5c0025f69a79397a9b1c0313017a6fe4071c

**Comment by tinyjin** [comment]
> it'd be nice if fallback would point to the concrete version from unpkg instead of always latest. Otherwise fallback will still break in production on releases.

Agree, will fix it soon. You can follow up this [issue](https://github.com/thorvg/thorvg.web/issues/85).

**Comment by gyzerok** [comment]
@tinyjin when trying to import it with vite like so `@thorvg/lottie-player/dist/thorvg-wasm.wasm?url` I am getting `Missing "./dist/thorvg-wasm.wasm" specifier in "@thorvg/lottie-player" package [plugin vite:dep-scan]`

**Comment by gyzerok** [comment]
Seems like what you are missing is this in `package.json`:
```json
  "exports": {
    ".": {
      "import": "./dist/lottie-player.cjs.js",
      "require": "./dist/lottie-player.esm.js"
    },
    "./dist/thorvg-wasm.wasm": {
      "import": "./dist/thorvg-wasm.wasm",
      "require": "./dist/thorvg-wasm.wasm"
    }
  },
```

**Comment by tinyjin** [comment]
@gyzerok 
Can you try this?
```ts
import wasmUrl from "../node_modules/@thorvg/lottie-player/dist/thorvg-wasm.wasm"
```

I think WASM binary file doesn't have to be handled by module bundler. Thus that is easy way to go.

**Comment by gyzerok** [comment]
@tinyjin will try it out, but I think it does, because multiple deployments have to have different versions and vite adds content hash to the file during builds. This way there are no conflicts between versions of the `wasm` file. Otherwise the problem would be similar to pointing internally to the latest unpkg version

**Comment by gyzerok** [comment]
@tinyjin I've tried it and getting the same error. I think without the `exports` bit in `package.json` modern module system considers it "internal". I've checked also with adding `exports` and then it works

**Comment by gyzerok** [comment]
@tinyjin sorry for the ping, but it seems like you've missed my last comment

**Comment by tinyjin** [comment]
@gyzerok Sorry for being late. Thanks for letting us know!

We'll review and support WASM entry export : )

**Comment by gyzerok** [comment]
Awesome, thank you!
